[kimberlybp] iP

kimberlybp opened the pull request:
# Sirius
> "Time is the scarcest resource and unless it is managed nothing else can be managed." - Peter Drucker ([source](https://www.brainyquote.com/quotes/peter_drucker_122188))

Sirius is a CLI app that helps you manage your time. It's,
- text-based
- easy to use
- ~~FAST~~ SUPER FAST to use

## In order to use it..
1. Download the JAR file from [here](https://github.com/kimberlybp/ip/releases/download/A-Jar/ip.jar)
2. Double-click it
3. Add your tasks
4. **Let Sirius handle the rest!** 😎

## Features
- [x] List all your tasks
- [x] Manage Todos, Deadlines and Events
- [ ] Reminders (Coming soon!)

### Devs feel free and play around with the code
Here's a preview of the `main` method: 
```java
public static void main(String[] args) {
    new Duke(PATH).run();
}
```

Commit:
Add Gradle support

kimberlybp committed:
Add Level 1. Greet, Echo, Exit

kimberlybp committed:
Add Level 2. Add, List

kimberlybp committed:
Add Level 3. Mark as Done

kimberlybp committed:
Add Level 4. ToDos, Events, Deadlines

kimberlybp committed:
Add A-TextUiTesting

kimberlybp committed:
Add Level 5. Handle Errors

kimberlybp committed:
Add Level 6. Delete

kimberlybp committed:
Add Level 7. Save

kimberlybp committed:
Add  Level 8. Dates and Times

kimberlybp committed:
Merge branch 'branch-Level-7'

kimberlybp committed:
Merge branch 'branch-Level-8'

kimberlybp committed:
Add A-MoreOOP,  A-Enums

kimberlybp committed:
Divide classes into packages, minor fixes

kimberlybp committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Skeltons commented on src/main/java/commands/CommandType.java:
```diff
@@ -0,0 +1,23 @@
+        this.value = value;
+    }
+
+    public boolean equals(String input) {
```
Perhaps the methods should be called `isEqual`?

jinbesan replied:
I think `isEqual` is a good idea!

Skeltons commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,72 @@
+package duke;
+
+import commands.*;
+import exceptions.DukeException;
+import tasks.*;
```
Shouldn't wildcard imports be avoided?

Skeltons commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import tasks.*;
+
+import java.io.*;
```
Shouldn't wildcard imports be avoided?

Skeltons commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,55 @@
+            String taskDesc = strArr[2];
+
+            switch (type) {
+                case "T":
+                    initList.add(new Todo(taskDesc, isCompleted));
+                    break;
+                case "D":
+                    initList.add(new Deadline(taskDesc, isCompleted, strArr[3]));
+                    break;
+                case "E":
+                    initList.add(new Event(taskDesc, isCompleted, strArr[3], strArr[4]));
+                    break;
```
Shouldn't there be no indentation between switch and case?

Skeltons reviewed (commented):
I've noticed that your Ui class is not being used properly as some classes are still printing to console without calling the Ui class. Overall, your code looks good to me!

jinbesan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,49 @@
+import java.util.Scanner;
+
+public class Duke {
+    private static final String PATH = "data/Duke.txt";
```
Nice use of capitalization for constants

jinbesan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    public static Command parse(String input) throws DukeException {
+        String[] inputArr = input.split(" ", 2);
```
Perhaps we can better understand this String array variable by naming it `inputs` or `inputTokens`?

jinbesan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,72 @@
+                    throw new DukeException("Hey now.. The description of a deadline cannot be empty. >:(");
+                }
+                taskInput = inputArr[1];
+                String[] dArr = taskInput.split("/by", 2);
```
The naming of this array might not be immediately evident on what it is supposed to be, because of the short name. Perhaps we can use a more precise name?

jinbesan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    public List<Task> loadFile() throws Exception {
+        List<Task> initList = new ArrayList<>();
```
Perhaps we can rename this Collection variable to be `tasks` as it is a collection of Tasks?

kimberlybp committed:
Update build.gradle shadowJar

kimberlybp committed:
Add JUnit tests

kimberlybp committed:
Add JavaDoc comments

kimberlybp committed:
Update code to comply with coding standard

kimberlybp committed:
Add Level 9. Find

kimberlybp committed:
Merge branch 'branch-A-JavaDoc'

kimberlybp committed:
Merge branch 'branch-A-CodingStandard'

kimberlybp committed:
Merge branch 'branch-Level-9'

kimberlybp committed:
Add JavaDoc comments for Level-9

kimberlybp committed:
Add checkstyle scripts, clear checkstyle errors

kimberlybp committed:
Merge branch 'branch-A-CheckStyle'

kimberlybp committed:
Fix data txt file not being created if no existing txt is found

kimberlybp committed:
Add Level 10 GUI

kimberlybp committed:
Merge branch 'branch-Level-10'

kimberlybp committed:
Fix text formatting

Previous duke CLI formatting had tabs included, but GUI update removed tabs in formatting, however some strings being returned still has tabs included

Remove remaining unnecessary tab escapes

To display text returned to GUI without unnecessary tabs to align text next to image

kimberlybp committed:
Update build.gradle

Previous build.gradle file had duke.Duke as mainClassName, however due to GUI addition to duke, it needs to be updated to duke.Launcher to launch GUI

Update mainClassName to duke.Launcher 

For JAR files to build and launch the main method of the correct class

kimberlybp committed:
Add Assertions

kimberlybp committed:
Refactor to improve code quality

Code can be further improved by applying code quality guidelines

Refactor and apply guidelines such as SLAP, KISS. Also rename certain variables for better readability

kimberlybp committed:
Merge pull request #2 from kimberlybp/branch-A-Assertions

Add Assertions

kimberlybp committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

kimberlybp committed:
Merge pull request #3 from kimberlybp/branch-A-CodeQuality

Improve code quality

kimberlybp committed:
Update code to clear checkstyle errors

kimberlybp committed:
Add priority param to Tasks

kimberlybp committed:
Merge branch 'branch-C-Extension'

kimberlybp committed:
Update Javadocs

kimberlybp committed:
Update package and chatbot class name

kimberlybp committed:
Fix MarkCommand response message

kimberlybp committed:
Update personality for duke

kimberlybp committed:
Merge branch 'branch-A-Personality'

kimberlybp committed:
Change DukeException to NookException

kimberlybp committed:
Update GUI to follow animal crossing theme

kimberlybp committed:
Clear checkstyle errors

kimberlybp committed:
Merge branch 'branch-A-BetterGui'

kimberlybp committed:
Add more error handling and improve existing error handling for Storage

kimberlybp committed:
Merge branch 'branch-A-MoreErrorHandling'

kimberlybp committed:
Add main window icon

kimberlybp committed:
Add placeholder message for List Command if no tasks

kimberlybp committed:
Enhance error handling in Parser to handle more edge cases

kimberlybp committed:
Add Help Command

kimberlybp committed:
Clear checkstyle errors

kimberlybp committed:
Update README.md, add Ui picture

kimberlybp committed:
Update Ui jpeg to png

kimberlybp committed:
Update README.md
